spring-integration-amqp/src/main/java/org/springframework/integration/amqp/channel/PollableAmqpChannel.java

```diff
@@ -322,12 +323,12 @@ public boolean removeInterceptor(ChannelInterceptor interceptor) {
 	}
 
 	@Override
-	public @Nullable ChannelInterceptor removeInterceptor(int index) {
+	public ChannelInterceptor removeInterceptor(int index) {
 		ChannelInterceptor interceptor = super.removeInterceptor(index);
 		if (interceptor instanceof ExecutorChannelInterceptor) {
 			this.executorInterceptorsSize--;
 		}
-		return interceptor;
+		return Objects.requireNonNull(interceptor);
```

Comment:
Remove this `requireNonNull` and change contract of that `super.removeInterceptor()`.
I agree it cannot be null, so let's fix that `AbstractMessageChannel.ChannelInterceptorList.remove(index)`